### PR TITLE
Start smoke test backend and collector only once

### PR DESCRIPTION
**Pull request by laurit**
Start backend and collector once for smoke tests and reuse it for all tests that run on the same worker. This should make smoke test runs a bit faster.

### CODE REVIEW

**Review by iNikem — APPROVED** [review]

File: smoke-tests/src/test/groovy/io/opentelemetry/smoketest/SmokeTest.groovy [hunk 1]
```diff
@@ -243,4 +219,59 @@ abstract class SmokeTest extends Specification {
...
+
+    int getMappedPort(int originalPort) {
+      return backend.getMappedPort(originalPort)
+    }
+
+    def cleanup() {
+    }
+
+    // currently unused
+    def stop() {
```
Comment (mateuszrzeszutek) [inline smoke-tests/src/test/groovy/io/opentelemetry/smoketest/SmokeTest.groovy:271]: Nit: consider adding a shutdown hook for stoping those docker containers

File: smoke-tests/src/test/groovy/io/opentelemetry/smoketest/SmokeTest.groovy [hunk 2]
```diff
@@ -243,4 +219,59 @@ abstract class SmokeTest extends Specification {
...
+    static Backend getInstance() {
+      return INSTANCE
+    }
+
+    def setup() {
+      // we start backend & collector once for all tests
+      if (started) {
+        return
+      }
+      started = true
```
Comment (mateuszrzeszutek) [inline smoke-tests/src/test/groovy/io/opentelemetry/smoketest/SmokeTest.groovy:241]: Do smoke tests run concurrently? If so, consider using an `AtomicBoolean` instead (and making the method synchronized)

**Review by mateuszrzeszutek — APPROVED** [review]

File: smoke-tests/src/test/groovy/io/opentelemetry/smoketest/SmokeTest.groovy (hunk 2 above)
Comment (laurit) [reply smoke-tests/src/test/groovy/io/opentelemetry/smoketest/SmokeTest.groovy:241]: My understanding is that tests run serially within a process, there can be multiple process, but they don't share anything. It is possible to make junit run tests in parallel with multiple threads but we don't use it.

File: smoke-tests/src/test/groovy/io/opentelemetry/smoketest/SmokeTest.groovy (hunk 1 above)
Comment (laurit) [reply smoke-tests/src/test/groovy/io/opentelemetry/smoketest/SmokeTest.groovy:271]: shutdown hook added

**Review by trask — APPROVED** [review]

**Pull request merged**